aea/protocols/generator/common.py

```diff
@@ -286,25 +301,23 @@ def _includes_custom_type(content_type: str) -> bool:
     :param content_type: the content type
     :return: Boolean result
     """
-    if content_type.startswith("Optional"):
-        sub_type = _get_sub_types_of_compositional_types(content_type)[0]
-        result = _includes_custom_type(sub_type)
-    elif content_type.startswith("Union"):
-        sub_types = _get_sub_types_of_compositional_types(content_type)
-        result = False
-        for sub_type in sub_types:
-            if _includes_custom_type(sub_type):
-                result = True
-                break
-    elif (
-        content_type.startswith("FrozenSet")
-        or content_type.startswith("Tuple")
-        or content_type.startswith("Dict")
-        or content_type in PYTHON_TYPE_TO_PROTO_TYPE.keys()
-    ):
-        result = False
-    else:
-        result = True
+    for type_ in [
+        "Union",
+        "Optional",
+        "pt:set",
+        "pt:list",
+        "Dict",
+        "Tuple",
+        "FrozenSet",
+    ]:
+        if content_type.startswith(type_):
+            sub_types = _get_sub_types_of_compositional_types(content_type)
+            for sub_type in sub_types:
+                if _includes_custom_type(sub_type):
+                    return True
+            return False
+
+    result = content_type not in PYTHON_TYPE_TO_PROTO_TYPE.keys()
```

Comment:
You are changing the purpose of this function. This used to work only with pythonic types and not specification types (`pt:`, `ct:`, etc). Is this needed now?
If it is needed to work also with specification types, you have not included `pt:dict`, `pt:union`, `pt:optional`.